tests/test_utils.py

```diff
@@ -184,15 +184,51 @@ def test_get_repository_config_missing(tmpdir):
     assert utils.get_repository_from_config(pypirc, "pypi") == exp
 
 
-def test_get_repository_config_invalid_url(tmpdir):
+def test_get_repository_config_invalid_scheme(tmpdir):
     """
-    Test if we get an URL without a protocol
+    Test if we get an URL with a invalid scheme
+    """
+
+    import re
+
+    pypirc = os.path.join(str(tmpdir), ".pypirc")
+
+    with pytest.raises(
+        exceptions.UnreachableRepositoryURLDetected,
+        match=re.escape(
+            r"Repository URL has an invalid scheme. "
+            r"scheme was required to be one of ['http', 'https'] but was 'ftp'"
+        ),
```

Comment:
I had to look up [`re.escape`](https://docs.python.org/3/library/re.html#re.escape); good to know. That said, looks like you could accomplish the same thing without the import:

```suggestion
        match=(
            r"Repository URL has an invalid scheme. "
            r"scheme was required to be one of \['http', 'https'\] but was 'ftp'"
        ),
```

Which would make this question moot: why is `import re` in the test, and not at the top of the module with the other imports?

Reply:
Thanks, this was able to get rid of `import re`